protocol/contracts/v2/TreasuryPriceIndexOracle.sol

```diff
@@ -46,40 +51,47 @@ contract TreasuryPriceIndexOracle is ITreasuryPriceIndexOracle, TempleElevatedAc
      */
     uint256 public override maxTreasuryPriceIndexDelta;
 
+    /**
+     * @notice The maximum allowed TPI change on any single `setTreasuryPriceIndex()`, in absolute terms.
+     * @dev Used as a bound to avoid unintended/fat fingering when updating TPI
+     */
+    uint32 public override maxTreasuryPriceIndexTargetDateDelta;
+
     constructor(
         address _initialRescuer,
         address _initialExecutor,
         uint96 _initialTreasuryPriceIndex,
         uint256 _maxTreasuryPriceIndexDelta,
-        uint32 _cooldownSecs
+        uint32 _maxTreasuryPriceIndexTargetDateDelta
     ) TempleElevatedAccess(_initialRescuer, _initialExecutor)
     {
         tpiData = TpiData({
             currentTpi: _initialTreasuryPriceIndex,
-            previousTpi: _initialTreasuryPriceIndex,
+            targetTpi: _initialTreasuryPriceIndex,
             lastUpdatedAt: uint32(block.timestamp),
-            cooldownSecs: _cooldownSecs
+            targetDate: uint32(block.timestamp),
+            tpiSlope: 1,
+            increaseInTargetTpi: true
         });
         maxTreasuryPriceIndexDelta = _maxTreasuryPriceIndexDelta;
+        maxTreasuryPriceIndexTargetDateDelta = _maxTreasuryPriceIndexTargetDateDelta;
     }
 
     /**
      * @notice The current Treasury Price Index (TPI) value
      * @dev If the TPI has just been updated, the old TPI will be used until `cooldownSecs` has elapsed
      */
     function treasuryPriceIndex() public override view returns (uint96) {
-        return (block.timestamp < (tpiData.lastUpdatedAt + tpiData.cooldownSecs))
-            ? tpiData.previousTpi  // use the previous TPI if we haven't passed the cooldown yet.
-            : tpiData.currentTpi;  // use the new TPI
-    }
-
-    /**
-     * @notice Set the number of seconds to elapse before a new TPI will take effect.
-     */
-    function setTpiCooldown(uint32 cooldownSecs) external override onlyElevatedAccess {
-        emit TpiCooldownSet(cooldownSecs);
-        tpiData.cooldownSecs = cooldownSecs;
-    }
+        if (block.timestamp >= tpiData.targetDate) {
```

Comment:
pull comment from direct commit

> Small gas optimisation:
> 
> uint32 _targetDate = tpiData.targetDate;
> And then use that local variable within. Each public variable read costs more gas than just a variable on the stack - so worth it if you access more than once in a function.

Reply:
ACK - will adjust. Had blinkers on with this being a view but didn't consider usage in other contracts.

Reply:
addressed in https://github.com/TempleDAO/temple/pull/1096/commits/7effc9f07f0365a4f88fa4fcc5884d0f05ef693a